objects/oplayercontroller/Step_0.gml

```diff
@@ -163,7 +166,90 @@ else
 	runBackwardTimer++;
 }
 
-show_debug_message(state);
+// If the run button and special button are pressed within 4 frames of each other, activate rush cancel
+// Also works with double tap forward, in which case the leniency is 7 frames
+if ((((runButton || special) && pressSpecialButtonTimer <= 4 && holdRunButtonTimer <= 4) 
+	|| (pressSpecialButtonTimer <= 7 && holdForwardTimer <= 7 && runningForward && !runButton))
+	&& state != eState.BEING_GRABBED // Prevent Rush Cancels during any of these states...
+	&& state != eState.THROW_TECH
+	&& state != eState.HURT
+	&& state != eState.LAUNCHED
+	&& state != eState.KNOCKED_DOWN
+	&& state != eState.GETUP
+	&& state != eState.BLOCKING
+	&& hitstun <= 0 && blockstun <= 0 && !FAvictim // and when the player gets hit...
+	&& superMeter >= 50 // and if the player doesn't have enough meter...
+	&& !rcActivated // and if the player already activated Rush Cancel and it hasn't completed yet.
+	&& !rcBuffer)
+{
+	pressSpecialButtonTimer = 8;
+	holdRunButtonTimer = 8;
+	if (opponent != noone && opponent.activateFreeze)
+	{
+		// 1 frame buffer to determine if both players activated RC at the same time
+		if (opponent.rcActivated && opponent.rcFreezeTimer > 1)
+		{
+			rcActivated = true;
+			rcFreezeTimer = 0;
+			superMeter -= 50;
+			projectileInvincible = true;
+			global.hitstop = 0;
+			activateFreeze = true;
+			global.freezeTimer = true;
+			state = eState.SCREEN_FREEZE;
+			instance_create_layer(x, y, "Instances", oRushCancel);
+		}
+		else
+		{
+			rcBuffer = true;
+			rcBufferTimer = 0;
+			rcBufferInterval = 30;
+		}
+	}
+	else if (opponent != noone && opponent.isGrabbed) // Activates buffer when grabbing

```

Comment:
Instead of using opponent.isGrabbed, you can use heldOpponent != noone, since heldOpponent stores the ID of the player we are grabbing.